### DISCUSSION TITLE
[Error Deployment] I cant't access my API from Custom Server Express JS when Deploy with Vercel or Netlify

### DISCUSSION THREAD

**Discussion by hialdev** [vercel/next.js#45110] [Help]
Hi, I'm new to next js

previously I learned React JS as front end and Express JS as backend, it requires 2 separate servers

I see that next js can run it on the same server, React as the front end and can also make APIs in it

I studied guided by chat.open.ai and read tutorials on the internet,

I tried my first learning project from UI Design to Next JS, I decided to use custom server express js because I had studied it before

I managed to get **everything running fine on localhost (xampp Mysql database)** with run command : _npm run server (node config/server.js)_

I also **created an Api outside of the pages/api folder to be precise in config/api/route-name.js**

and my **custom server is located in config/server.js**

You can see my source code at:
github.com/hialdev/devtektif

The problem is **when I deploy to Vercel my data fetch api doesn't work, so it's blank when viewed on live site, same thing happens on netlify**

Please help me, how can I deploy my application so that it runs well****

on localhost :
![image](https://user-images.githubusercontent.com/73730211/213787880-298d2cc3-3f80-482c-8313-30c4a556f439.png)

deploy on vercel :
![image](https://user-images.githubusercontent.com/73730211/213788054-e2dcc4c6-aef8-4667-a1f1-e8a29bbd9f50.png)
https://devtektif.vercel.app/

deploy on netlify :
![image](https://user-images.githubusercontent.com/73730211/213788253-c90996ae-0c06-434d-bfaa-9455ad8159e7.png)
https://63cae5573d2957004e8141c3--singular-malasada-bb05d8.netlify.app/

I am very grateful for your help :)

**Comment by devjiwonchoi** [comment]
I guess you haven't set up an actual database on any server yet?
Dump your MySQL DB to a hosted server and connect it via ORM (Sequelize, Prisma, etc.).

Note that `xampp` is for your local machine, not going to be reachable as an API right away.

P.S. Vercel supports serverless DB but is PSQL - https://vercel.com/docs/storage/vercel-postgres
